### ISSUE TITLE
Compiling parser from scratch produces ABI version 9 even when tools are all at version 13 or 14.

### ISSUE THREAD

**Issue by TomLebeda** [nvim-treesitter/nvim-treesitter#5302]
Hello, 

I have made myself an ABNF parser as described [here](https://tree-sitter.github.io/tree-sitter/creating-parsers), but when trying to add it to Neovim (following [Adding parsers](https://github.com/nvim-treesitter/nvim-treesitter#adding-parsers)) 
I encounter following problem after running `:TSInstall abnf`: 

```
Could not load parser for abnf: "/usr/sha...esitter/language.lua:99: ABI version mismatch for ~/.local/share/nvim/site/pack/packer/start/nvim-treesitter/parser/abnf.so: supported between 13 and 14, found 9"
```

The parser itself works, I can parse and highlight files using the CLI tool without problems. 

I have been trying to solve this for two days now, but all the issues on this topic are quite old and nothing described worked for me. 

It seems to me as if the installing process creates the `.so` file with ABI version 9 no matter what I try. But the tree-sitter tooling I have supports versions 13 and 14. 

Any ideas what I might be doing wrong? I suspect it's something badly configured, but I have no idea where to look, everything seems to me like it should work. 

```
$ nvim --version
NVIM v0.9.1
Build type: Release
LuaJIT 2.1.0-beta3
```

```
$ tree-sitter -V
tree-sitter 0.20.8 (0c49d6745b3fc4822ab02e0018770cd6383a779c)
```

(after running `tree-sitter generate -h` I see that the `--abi` flag has default value 14)

after running `:checkhealth nvim-treesitter`:
```
nvim-treesitter: require("nvim-treesitter.health").check()

Installation ~
- OK `tree-sitter` found 0.20.8 (0c49d6745b3fc4822ab02e0018770cd6383a779c) (parser generator, only needed for :TSInstallFromGrammar)
- OK `node` found v20.5.0 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- OK `gcc` executable found. Selected from { "gcc" }
  Version: gcc (GCC) 13.2.1 20230801
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "x86_64",
  release = "6.3.13-1-MANJARO",
  sysname = "Linux",
  version = "#1 SMP PREEMPT_DYNAMIC Tue Jul 11 22:21:57 UTC 2023"
} ~

Parser/Features         H L F I J
  - abnf                . . . . .
  - c                   ✓ ✓ ✓ ✓ ✓
  - cpp                 ✓ ✓ ✓ ✓ ✓
  - go                  ✓ ✓ ✓ ✓ ✓
  - julia               ✓ ✓ ✓ ✓ ✓
  - latex               ✓ . ✓ . ✓
  - lua                 ✓ ✓ ✓ ✓ ✓
  - nu                  ✓ ✓ ✓ . ✓
  - python              ✓ ✓ ✓ ✓ ✓
  - rust                ✓ ✓ ✓ ✓ ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~

```

**Comment by clason** [comment]
That is weird indeed; are you sure it's not finding the five year old parser for some reason? That's the only thing I can think of.

**Comment by TomLebeda** [comment]
I don't think so. I tried to clean the cache, completely delete the old repo from my computer. I also changed the url in my `init.lua` to point to the directory of my new parser. 

I tried to search my whole computer (from root directory) with catfish for `abnf`. After the eon it took, I deleted all the traces of the old repo that it found. So unless there is some persistent caching in RAM, it should even know some old repo ever existed. 

So if there is some left-over link to the old repo somewhere, I have no idea where it could be. At this point the only thing I didn't try is completely reinstall Neovim.

**Comment by clason** [comment]
Or maybe a very, very old tree-sitter CLI version in your PATH that Neovim picks up over the one you expect?

**Comment by TomLebeda** [comment]
I see `tree-sitter-cli` installed via pacman that has version 0.20.8-4, which is new one. And `tree-sitter` via pacman has version 0.20.10-1. 

I'm not sure how to verify it that's the one that is used. I don't see any other version installed and console always uses these new ones.

**Comment by clason** [comment]
To be clear, I have no idea what's going on. Nvim-treesitter does not specify any ABI, and if you have a current tree-sitter version, you should not even be able to create a n ABI 9 parser. (Unless the generated files in your repo are that old? Then you just cannot use them with Neovim; you have to generate the parser.c locally.)

**Comment by TomLebeda** [comment]
Good catch, the 0.20.10 is indeed suspicious, although that number is only in the Pacman GUI and it also sais that it was installed as a dependency for another package (it says required by neovim). I will try to remove it and see what happens. 

And about the files - all of the files in my tree-sitter-abnf directory are generated this week, I followed the tutorial "Creating parsers" in the TreeSitter docs.

**Comment by clason** [comment]
If it's your repo and you commit the parser.c, make sure it's created with the 0.20.8 version. That should fix the error.

**Comment by clason** [comment]
> I am pretty sure that the tree-sitter generate was called correctly and the parser.c was created by the 0.20.8 version.

I'm pretty sure that's not the case, as the current version of tree-sitter CLI can only create ABI 13 or 14.

> Should I close this issue?

I'd say yes, as I don't see how this can conceivably be a nvim-treesitter issue. We can always reopen if you find out otherwise later.

Last thing you could try in way of a workaround is to add `requires_generate_from_grammar = true` to your custom parser config; that should bypass the `parser.c` in the repo.
